### PR TITLE
Pm search for packages - versions updates

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

Package versions as a drop-down list allowing the user to install a specified version. Resurrected a feature (pre-2.19) allowing you to install the package in a specific folder. 

#### UI Changes
versions update

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

- versions now a drop-down allowing to install selected version instead of always installing the latest
- install to selected folder option resurrection
- UI design updates as per Figma
- 
### Reviewers

@QilongTang 
@reddyashish 

### FYIs

@Amoursol

### CODE REVIEW

**Comment by dnenov** [comment]
2 failing tests already fixed in [this PR](https://github.com/DynamoDS/Dynamo/pull/14352). I will mark this one as a `draft` for the time being, until I can merge the updated Master with the other branch, if that's OK.

**Comment by dnenov** [comment]
> Since #14352 is merged, this PR is now ready for review I assume?

Yes, this one should now be good to go!

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@dnenov Two regressions since you touched search element:
[Dynamo.Tests.Search.SearchElements.SearchElementBaseTest.TestNodeModelSearchElement](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/11135/testReport/junit/Dynamo.Tests.Search.SearchElements/SearchElementBaseTest/TestNodeModelSearchElement/)
[Dynamo.PackageManager.Wpf.Tests.PackageManagerSearchElementViewModelTests.TestPackageManagerSearchElementCanInstall](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/11135/testReport/junit/Dynamo.PackageManager.Wpf.Tests/PackageManagerSearchElementViewModelTests/TestPackageManagerSearchElementCanInstall/)

File: src/DynamoPackages/PackageManagerSearchElement.cs [hunk 1]
```diff
@@ -42,6 +42,8 @@ public int Votes
         public int UsedBy { get { return this.Header.used_by.Count; } }
         public string LatestVersion { get { return Header.versions[Header.versions.Count - 1].version; } }
         public string LatestVersionCreated { get { return Header.versions[Header.versions.Count - 1].created; } }
+        public string SelectedVersion { get; set; }
```
Comment (QilongTang) [inline src/DynamoPackages/PackageManagerSearchElement.cs:45]: theoretically this should be on view model layer but I guess such layer does not exist for search element...

Comment (dnenov) [reply src/DynamoPackages/PackageManagerSearchElement.cs:45]: Hmm .. no, not really :/ Of course, you are right, we shouldn't be binding directly to the Model. Strange, the existing bindings go straight to the model. Should I refactor those as well? Or not touch the existing ones.

Comment (QilongTang) [reply src/DynamoPackages/PackageManagerSearchElement.cs:45]: hmm, we can leave that as a ToDo in a new PR if that is OK

Comment (dnenov) [reply src/DynamoPackages/PackageManagerSearchElement.cs]: Moved the `SelectedVersion` to the VM, we will fix the rest in a separate PR as you said, @QilongTang !

**Comment by QilongTang** [comment]
Still one regression [Dynamo.PackageManager.Wpf.Tests.PackageManagerSearchElementViewModelTests.TestPackageManagerSearchElementCanInstall](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/11146/testReport/junit/Dynamo.PackageManager.Wpf.Tests/PackageManagerSearchElementViewModelTests/TestPackageManagerSearchElementCanInstall/)

**Comment by QilongTang** [comment]
Triggered PR check manually at https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/11148/

**Pull request merged**